File: source/python/dummy.py
# ---------------------------------------------------------------------
#
# Copyright (C) 2016 by the deal.II authors
#
# This file is part of the deal.II library.
#
# The deal.II library is free software; you can use it, redistribute
# it, and/or modify it under the terms of the GNU Lesser General
# Public License as published by the Free Software Foundation; either
# version 2.1 of the License, or (at your option) any later version.
# The full text of the license can be found in the file LICENSE at
# the top level of the deal.II distribution.
#
# ---------------------------------------------------------------------

"""This a dummy module that contains a global function and a class."""

__all__ = ['Dummy', 'bar']

def bar():
    """Print bar."""
    print('bar')

class Dummy:
    """This a dummy class that has only one member function."""
    
    def foo(self):
        """Print foo."""
        print('foo')

File: source/dummy.cc
// -----------------------------------------------------------------------------
//
// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception OR LGPL-2.1-or-later
// Copyright (C) 2013 - 2020 by the deal.II authors
//
// This file is part of the deal.II library.
//
// Detailed license information governing the source code and contributions
// can be found in LICENSE.md and CONTRIBUTING.md at the top level directory.
//
// -----------------------------------------------------------------------------

/*
 * Workaround for a bug in the Xcode generator.
 *
 * This file contains a dummy global symbol to trigger the link phase in
 * the generated Xcode project.
 */

const int global_symbol_42{42};
void
use_global_symbol_42()
{
  (void)global_symbol_42;
}

/**
 * If we are running the contrib/utilities/run_clang_tidy.sh script, we
 * generate a header file allheaders.h that includes all deal.II
 * headers. Include the file here so that all headers are being checked.
 */
#ifdef CLANG_TIDY
#  include <deal.II/allheaders.h>
#endif
